stores/src/main/java/dataBase/DataBaseStorePopulator.java

```diff
@@ -0,0 +1,149 @@
+                    rs.next();
+                    List<Product> productTable = RandomStorePopulator.createListProduct(category.getName());
+                    for (Product product : productTable) {
+                        String SQL1 = "INSERT INTO PRODUCT (productName, productPrice, productRate, category_id)" + "VALUES( ?, ?, ?, ?)";
+                        preparedStatement = dataBaseConnect.connection().prepareStatement(SQL1);
+                        preparedStatement.setString(1, product.getName());
+                        preparedStatement.setDouble(2, product.getPrice());
+                        preparedStatement.setDouble(3, product.getRate());
+                        preparedStatement.setLong(4, rs.getLong(1));
+                        preparedStatement.executeUpdate();
+                    }
+                }
+                fillCategoryDB();
+            }
+
+            statement.close();
+            dataBaseConnect.connection().close();
+        } catch (SQLException se) {
+            throw new RuntimeException("Database access error", se);
+        } catch (Exception e) {
+            throw new RuntimeException("Error", e);
+        } finally {
+            try {
+                if (statement != null) statement.close();
+            } catch (SQLException se2) {
+                throw new RuntimeException("Database access error", se2);
+            }
+            try {
+                if (dataBaseConnect.connection() != null) dataBaseConnect.connection().close();
+            } catch (SQLException se) {
+                throw new RuntimeException("Database access error", se);
+            }
+        }
+//        System.out.println(categoryListDB);
+        return categoryListDB;
+    }
+
+    private void fillCategoryDB() throws SQLException {
+        categoryListDB = new ArrayList<>();
+        String sql = "SELECT name, id FROM CATEGORY ";
+        ResultSet rs = statement.executeQuery(sql);
+        Category category = null;
+        while (rs.next()) {
+            String categoryName = rs.getString("name");
+            switch (categoryName) {
+                case ("food"):
+                    category = new Food();
+                    break;
+                case ("book"):
+                    category = new Book();
+                    break;
+                case ("animal"):
+                    category = new Animal();
+                    break;
+            }
+            category.setId(rs.getInt("id"));
+            categoryListDB.add(category);
+        }
+
+        for (Category categoryDB : categoryListDB) {
+            categoryDB.setListProduct(fillProdListDB(categoryDB.getId()));
+        }
+    }
+
+    private List<Product> fillProdListDB(int id) {
+        List<Product> productListDB = new ArrayList<>();
+        String sql1 = String.format("SELECT * FROM PRODUCT WHERE category_id = '%d' ", id);
+        ResultSet rs1 = null;
+        try {
```

Comment:
very very strange code to process result set, why not just while(rs.next) {Product p new Product(rs.getString(...